packages/transactions/src/builders.ts

```diff
@@ -1293,10 +1295,14 @@ export async function callReadOnlyFunction(
 
   const options = Object.assign(defaultOptions, readOnlyFunctionOptions);
 
-  const { contractName, contractAddress, functionName, functionArgs, senderAddress } = options;
+  const { contractName, contractAddress, functionName, functionArgs, senderAddress, tip } = options;
 
   const network = StacksNetwork.fromNameOrNetwork(options.network);
-  const url = network.getReadOnlyFunctionCallApiUrl(contractAddress, contractName, functionName);
+  let url = network.getReadOnlyFunctionCallApiUrl(contractAddress, contractName, functionName);
+
+  if (typeof tip === 'number') {
+    url = `${url}?tip=${tip}`;
```

Comment:
Blindly appending `?` to an opaque URL is bad idea. In this case, the above function that generates the url can already add query params:
https://github.com/hirosystems/stacks.js/blob/266f473276f209a2cdd7044313a64c5e40a93455/packages/network/src/index.ts#L76-L78

So you could end up with a broken query string like `myurl.com/endpoint?thing1?thing2`.

The URL construction should probably all be contained within that existing function, or at least use the `URL` class to safely append query params.